tests/config/source/test_discover.py

```diff
@@ -10,7 +10,7 @@
 
 def out_no_src(path: Path) -> str:
     return (
-        f"ROOT: No tox.ini or setup.cfg or pyproject.toml found, assuming empty tox.ini at {path}\n"
+        f"ROOT: No tox.ini or tox.toml or setup.cfg or pyproject.toml found, assuming empty tox.ini at {path}\n"
```

Comment:
I'm not sure I agree with the need for a dedicated TOML file.  I am currently -1 on adding that.

Reply:
I am in favour of defaulting to `pyproject.toml` but still allowing user to define an alternative config file to load. Some might find a need for it (like CI/CD, or custom user modified one when they cannot merge their changes in the default one).

I would say that even with alternative filename, I would still expect it to follow the same sections as the `pyproject.toml`, just to keep code simple.

Reply:
I went through the docs for several non-build tools I know (excluded those which do not support `pyproject.toml` at all) to see if the support custom TOML file. Here are results:

| Tool | Custom TOML file | Custom TOML root |
| ---- | --------------------- | ------------------ |
| mypy | none | |
| ruff | `ruff.toml` | none |
| bump-my-version | `.bumpversion.toml` | `tool.bumpversion` or `bumpversion` |
| isort | none | |
| black | none | |
| codespell | none | |
| coverage | none | |
| towncrier | `towncrier.toml` | `tool.towncrier` |

I'd say custom TOML file is not that rare. Custom root in TOML file seems to vary.